quipucords/fingerprinter/formatters.py

```diff
@@ -31,3 +35,57 @@ def convert_architecture(architecture):
     if not architecture_map.get(architecture):
         return architecture
     return architecture_map.get(architecture)
+
+
+def str_or_none(value):
+    """Normalize value to be a string (or None when "empty")."""
+    if value is None:
+        return None
+    if isinstance(value, str) and value.strip() == "":
+        return None
+    return str(value).strip()
+
+
+def float_or_none(value):
+    """Convert value to float."""
+    with suppress(ValueError, TypeError):
+        return float(value)
+    return None
+
+
+def int_or_none(value):
+    """Convert value to int."""
+    try:
+        return int(value)
+    except TypeError:
+        return None
+    except ValueError:
+        ...
+
+    with suppress(ValueError):
+        return int(float_or_none(value))
+
+    return None
+
+
+def list_or_none(value):
```

Comment:
Maybe we should add some type hints so our editors might yell at us if we are using these wrongly.

```suggestion
def list_or_none(value: list) -> list | None:
```